### ISSUE TITLE
feat: Support Pub Key Encryption

### ISSUE THREAD

**Issue by enricobottazzi** [gaussian-dev/halo2-fhe#16]

**Comment by mahmudsudo** [comment]
can i take on this ?

**Comment by enricobottazzi** [comment]
Hi @mahmudsudo, I think this issue has already been by @Vishalkulkarni45. Would you be interested in taking #17. I can provide more guidance if needed.

**Comment by Vishalkulkarni45** [comment]
Yes as mentioned by @enricobottazzi ,I will take on this task and see it through to completion.

**Comment by 0xjei** [comment]
> As described [here](https://hackmd.io/@gaussian/r1W98Kqqa). It requires to extend the existing `sk_encryption_circuit`

Hey, I just wonder if we can get access to this hackmd file again @enricobottazzi says not found on my side. I'd like to keep this for future docs

**Comment by enricobottazzi** [comment]
@0xjei the hackmd file was transformed in the paper. So you can refer to the paper.

**Comment by 0xjei** [comment]
> @0xjei the hackmd file was transformed in the paper. So you can refer to the paper.

makes sense, thx
